interop-cats/jvm/src/test/scala/zio/interop/CatsZManagedSyntaxSpec.scala

```diff
@@ -3,334 +3,302 @@ package zio.interop
-      Resource.makeCase(Task { effects += x; () }) {
-        case (_, ExitCase.Error(_)) =>
-          Task { effects += x + 1; () }
-        case _ =>
-          Task.unit
-      }
-
-    val testCase = {
-      val managed: ZManaged[Any, Throwable, Unit] = res(1).toManagedZIO
-      managed.use(_ => ZIO.fail(new RuntimeException()).unit)
-    }
-
-    unsafeRun(testCase.orElse(ZIO.unit))
-    effects must be_===(List(1, 2))
-  }
-
-  def toManagedZIOFinalizersWhenDied = {
-    val effects = new mutable.ListBuffer[Int]
-    def res(x: Int): Resource[Task, Unit] =
-      Resource.makeCase(Task { effects += x; () }) {
-        case (_, ExitCase.Error(_)) =>
-          Task { effects += x + 1; () }
-        case _ =>
-          Task.unit
-      }
-
-    val testCase = {
-      val managed: ZManaged[Any, Throwable, Unit] = res(1).toManagedZIO
-      managed.use(_ => ZIO.die(new RuntimeException()).unit)
-    }
-
-    unsafeRun(testCase.sandbox.orElse(ZIO.unit))
-    effects must be_===(List(1, 2))
-  }
-
-  def toManagedZIOFinalizersExceptionAcquisition = {
-    val effects = new mutable.ListBuffer[Int]
-    def res(x: Int): Resource[Task, Unit] =
-      Resource.make(Task(effects += x) *> Task(throw new RuntimeException()).unit)(
-        _ => Task { effects += x + 1; () }
-      )
+  def spec =
+    suite("CatsZManagedSyntaxSpec")(
+      suite("toManaged")(
+        test("calls finalizers correctly when use is interrupted") {
+          val effects = new mutable.ListBuffer[Int]
+          def res(x: Int): Resource[CIO, Unit] =
+            Resource.makeCase(CIO.delay { effects += x; () }) {
+              case (_, ExitCase.Canceled) =>
+                CIO.delay { effects += x + 1; () }
+              case _ => CIO.unit
+            }
+
+          val testCase = {
+            val managed: ZManaged[Any, Throwable, Unit] = res(1).toManaged
+            managed.use(_ => ZIO.interrupt.unit)
+          }
+
+          unsafeRun(testCase.orElse(ZIO.unit))
+          assert(effects.toList)(equalTo(List(1, 2)))
+        },
+        test("calls finalizers correctly when use has failed") {
+          val effects = new mutable.ListBuffer[Int]
+          def res(x: Int): Resource[CIO, Unit] =
+            Resource.makeCase(CIO.delay { effects += x; () }) {
+              case (_, ExitCase.Error(_)) =>
+                CIO.delay { effects += x + 1; () }
+              case _ =>
+                CIO.unit
+            }
+
+          val testCase = {
+            val managed: ZManaged[Any, Throwable, Unit] = res(1).toManaged
+            managed.use(_ => ZIO.fail(new RuntimeException()).unit)
+          }
+
+          unsafeRun(testCase.orElse(ZIO.unit))
+          assert(effects.toList)(equalTo(List(1, 2)))
+        },
+        test("calls finalizers correctly when use has died") {
+          val effects = new mutable.ListBuffer[Int]
+          def res(x: Int): Resource[CIO, Unit] =
+            Resource.makeCase(CIO.delay { effects += x; () }) {
```

Comment:
`effects += x; ()` is really ugly :)

Reply:
Probably could be replaced by `.as(())`

Reply:
`.void`. But I actually prefer `; ()` in general since it's more efficient.

Reply:
Thank you for review 🙇 
Changed to `.void` / `.unit`
(Since this is in test code, it is probably more beneficial to have code easy to read than tuned in terms of performance.)